### PR TITLE
[Transform] Encapsulate plugin-level settings into a `TransformExtension` interface

### PR DESCRIPTION
**Pull request by przemekwitek**
This PR encapsulate plugin-level settings into a `TransformExtension` interface so that these settings can be injected easily into actions.

Marking `>non-issue` as there is no change in behavior.

### CODE REVIEW

**Review by droberts195 — APPROVED** [review]
LGTM

File: x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/Transform.java [hunk 1]
```diff
@@ -467,11 +479,17 @@ public String getFeatureDescription() {
...
     }
 
+    public TransformExtension getTransformExtension() {
+        return transformExtension;
+    }
+
+    @Deprecated
     public boolean includeNodeInfo() {
-        return true;
+        return transformExtension.includeNodeInfo();
```
Comment (droberts195) [inline x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/Transform.java:488]:
```suggestion
        return getTransformExtension().includeNodeInfo();
```

**Review by droberts195 — COMMENTED** [review]
I think it's best to make the 2 deprecated methods return the correct values for serverless, just in case somebody does use them somewhere.

File: x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/Transform.java (hunk 1 above)
Comment (przemekwitek) [reply x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/Transform.java]: Done.

File: x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/Transform.java [hunk 2]
```diff
@@ -467,11 +479,17 @@ public String getFeatureDescription() {
...
     }
 
+    @Deprecated
     public Settings getTransformInternalIndexAdditionalSettings() {
-        return Settings.EMPTY;
+        return transformExtension.getTransformInternalIndexAdditionalSettings();
```
[the remark this answers is not in the document]
Comment (przemekwitek) [reply x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/Transform.java]: Done.

**Pull request merged**